P2PEnabledConference.js

```diff
@@ -0,0 +1,899 @@
+     * @param {string} eventType the event type name
+     * @return {boolean} <tt>true</tt> to block or <tt>false</tt> to let through
+     * @private
+     */
+    _shouldBlockEvent (eventType) {
+        switch (eventType) {
+            case JitsiConferenceEvents.CONNECTION_INTERRUPTED:
+            case JitsiConferenceEvents.CONNECTION_RESTORED:
+                return this.p2pEstablished;
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * Starts new P2P session.
+     * @param {string} peerJid the JID of the remote participant
+     * @private
+     */
+    _startPeer2PeerSession(peerJid) {
+        if (this.deferredStartP2P) {
+            // Make not that the task has been executed
+            this.deferredStartP2P = null;
+        }
+        if (this.peerToPeerSession) {
+            logger.error("P2P session already started!");
+            return;
+        }
+
+        /*if (!peerJid) {
+            const peers = this.getParticipants();
+            const peerCount = peers.length;
+
+            // Start peer to peer session
+            if (peerCount > 0) {
+                peerJid = peers[0].getJid();
+            } else {
+                logger.error("No JID to start the P2P session with !");
+                return;
+            }
+        }*/
+
+        this.peerToPeerSession
+            = this.xmpp.connection.jingle.newJingleSession(
+                this.room.myroomjid, peerJid);
+        this.peerToPeerSession.setSSRCOwnerJid(this.room.myroomjid);
+        this.peerToPeerFakeRoom = this._createFakeRoom(true);
+
+        logger.info(
+            "Created new P2P JingleSession", this.room.myroomjid, peerJid);
+
+        this.peerToPeerSession.initialize(
+            true /* initiator */, this.peerToPeerFakeRoom, this.rtc);
+
+        // NOTE one may consider to start P2P with the local tracks detached,
+        // but no data will be sent until ICE succeeds anyway. And we switch
+        // immediately once the P2P ICE connects.
+        const localTracks = this.getLocalTracks();
+
+        logger.info("Adding " + localTracks + " to P2P...");
+        this.peerToPeerSession.addLocalTracks(localTracks).then(
+            () => {
+                logger.info("Added " + localTracks + " to P2P");
+                logger.info("About to send P2P 'session-initiate'...");
+                this.peerToPeerSession.invite();
+            },
+            (error) => {
+                logger.error("Failed to add " + localTracks + " to P2P", error);
+            });
+    }
+
+    /**
+     * Method when called will decide whether it's the time to start or stop the
+     * P2P session.
+     * @param {boolean} userLeftEvent if <tt>true</tt> it means that the call
+     * originates from the user left event.
+     * @private
+     */
+    _startStopP2PSession (userLeftEvent) {
+        if (this.options.config.disableAutoP2P) {
+            logger.info("Auto P2P disabled");
```

Comment:
It disables automatic switching between JVB and P2P and then it can be done manually using this methods: https://github.com/jitsi/lib-jitsi-meet/blob/peer-to-peer/P2PEnabledConference.js#L763. I use this property in torture tests to enforce switch from JVB to P2P at specific state - when user is video muted. I will add better description to the method and will describe this config option.